trigger.go

```diff
@@ -0,0 +1,99 @@
+package cosmosdb
+
+import (
+	"context"
+)
+
+type Trigger struct {
+	Resource
+	Id        string           `json:"id"`
+	Body      string           `json:"body"`
+	Operation TriggerOperation `json:"triggerOperation"`
+	Type      TriggerType      `json:"triggerType"`
+}
+
+type TriggerType string
+type TriggerOperation string
+
+type CollectionTriggers struct {
+	Rid      string    `json:"_rid,omitempty"`
+	Count    int32     `json:"_count,omitempty"`
+	Triggers []Trigger `json:"Triggers"`
+}
+
+//const (
+//	TriggerTypePost = TriggerType("Post")
+//	TriggerTypePre  = TriggerType("Pre")
+//
+//	TriggerOpAll     = TriggerType("All")
+//	TriggerOpCreate  = TriggerType("Create")
+//	TriggerOpReplace = TriggerType("Replace")
+//	TriggerOpDelete  = TriggerType("Delete")
+//)
+
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-trigger
+type TriggerCreateOptions struct {
+	Id        string           `json:"id"`
+	Body      string           `json:"body"`
+	Operation TriggerOperation `json:"triggerOperation"`
+	Type      TriggerType      `json:"triggerType"`
+}
+
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/replace-a-trigger
+type TriggerReplaceOptions struct {
+	Id        string           `json:"id"`
+	Body      string           `json:"body"`
+	Operation TriggerOperation `json:"triggerOperation"`
+	Type      TriggerType      `json:"triggerType"`
+}
+
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-trigger
+func (c *Client) CreateTrigger(ctx context.Context, dbName string, colName string,
+	trigOps TriggerCreateOptions, ops *RequestOptions) (*Trigger, error) {
```

Comment:
And the same for the `RequestOptions` parameter, I would like to replace it with explicit option struct to be a) consistent in the driver and b) make it easy (i.e. explicit) to use. With the `map[string]string` beneath the `RequestOptions` struct, you don't know what you can put in as a caller.